hypothesis-python/src/hypothesis/searchstrategy/recursive.py

```diff
@@ -101,9 +100,6 @@ def do_draw(self, data):
             except LimitReached:
                 if count == 0:
                     data.note_event(
-                        lazyformat(
-                            "Draw for %r exceeded max_leaves and had to be retried",
-                            self,
-                        )
+                        "Draw for %r exceeded max_leaves and had to be retried" % self
```

Comment:
Are you sure this is pointless? It might be, but I think it needs more of an argument that this is the case - the problem `lazyformat` solves is that the `repr` for strategies may be (very) expensive to evaluate, and we don't want to calculate it each time we call `note_event` unless we need to.